olclient/openlibrary.py

```diff
@@ -656,6 +656,49 @@ def save(self, comment):
                 url = self.OL.base_url + '/authors/%s.json' % self.olid
                 return self.OL.session.put(url, json.dumps(body))
 
+            def works(self, limit=50, offset=0):
+                """Returns a list of OpenLibrary Works associated with an OpenLibrary Author.
+
+                Args:
+                    olid (unicode) - OpenLibrary ID for author to search within
+                                    Open Library's database of authors to retrieve his Works.
+                    name (unicode) - name of an Author to search for within OpenLibrary.
+                    limit (integer) - number of Author's Works to return.
+                    offset (integer) - offset number to aid pagination.
+                Returns:
+                    A (list) of Works from the OpenLibrary associated with the
+                    Author.
+
+                Usage:
+                    >>> from olclient.openlibrary import OpenLibrary
+                    >>> ol = OpenLibrary()
+                    >>> ol.Author.get('OL39307A').works()
+                    or
+                    >>> ol.Author.get('OL39307A').works(limit=20)# to obtain the first 20 works of the author
+                    >>> ol.Author.get('OL39307A').works(limit=20, offset=20)# to obtain the next 20 works of the author
+                    or
+                    >>> author_obj = ol.Author.get(ol.Author.get_olid_by_name('Dan Brown'))
+                    >>> author_obj.works()
+                    or
+                    >>> ol.Author.get(ol.Author.get_olid_by_name('Dan Brown')).works()
+                """
+                path = '/authors/%s/works.json' % self.olid
+
+                if limit:
```

Comment:
This code block seems to implicitly require that `limit` exists because it is what adds `?` to the url path.

If the programmer calls this `works` method with `limit=None` then `?` will never get added to the url. If `limit` is `None` but `offset` is `not None` then the url becomes `openlibrary.org/authors/OL...A.json&offset=0` which will be a malformed query.

Perhaps we can make the following change:
```
WORKS_LIMIT = 50  # near top of file as a constant
...
def works(self, limit=WORKS_LIMIT, offset=0):
    limit = limit or WORKS_LIMIT  # this guarantees limit is never None or 0
```

Then we no longer need to check `if limit:`, we can always add `'/?limit=%s' % limit`

Reply:
Ah, this is a needful consideration that needs to be handled.
Thanks for pointing it out.

Resolved changes requested in the review in commit:[efe008a](https://github.com/internetarchive/openlibrary-client/pull/136/commits/efe008a4a32f511d349267b96a97a5d8ddfaf1f4)